cmd/subctl/upgrade.go

```diff
@@ -0,0 +1,271 @@
+		exit.OnError(upgradeRestConfigProducer.RunOnAllContexts(upgradeSubmariner, status))
+	}
+}
+
+// upgradeSubctl upgrades the local copy of subctl, if necessary.
+// Returns the path to the upgraded subctl if subctl was upgraded, nil if it wasn't.
+func upgradeSubctl(status reporter.Interface) (*string, error) {
+	// Default to downloading the latest version
+	targetVersionString := "latest"
+
+	// If the user hasn't specified a version, try to find the latest release on GitHub
+	if to == "" {
+		client := github.NewClient(nil)
+		latestRelease, _, err := client.Repositories.GetLatestRelease(context.TODO(), "submariner-io", "releases")
+
+		// If we can't determine the latest release, we'll force a download and delegate to get.submariner.io
+		if err == nil {
+			// Add the v prefix so that the version will match when compared in the new subctl
+			to = "v" + *latestRelease.TagName
+		}
+	}
+
+	if to == version.Version {
+		// Already running the right version
+		return nil, nil //nolint:nilnil // A nil command means subctl shouldn't be restarted.
+	}
+
+	if to != "" {
+		if len(to) > 1 && to[0:1] == "v" {
+			to = to[1:]
+		}
+
+		toVersion, err := semver.NewVersion(to)
+		if toVersion == nil {
+			return nil, status.Error(err, "Invalid target version")
+		}
+
+		// semver needs a dotted triplet, which is at least five characters;
+		// on development or unknown versions, assume we need to upgrade
+		if len(version.Version) >= 5 && version.Version[0:5] != "devel" {
+			currentVersion, err := semver.NewVersion(version.Version)
+			if currentVersion == nil {
+				return nil, status.Error(err, "Error parsing current subctl version")
+			}
+
+			if toVersion.LessThan(*currentVersion) || toVersion.Equal(*currentVersion) {
+				return nil, nil //nolint:nilnil // A nil command means subctl shouldn't be restarted.
+			}
+		}
+
+		targetVersionString = "v" + toVersion.String()
+	}
+
+	status.Start(fmt.Sprintf("Upgrading subctl from %s to %s, replacing %s", version.Version, targetVersionString, os.Args[0]))
+	defer status.End()
+
+	url := "https://get.submariner.io"
+
+	absolutePath, err := filepath.Abs(os.Args[0])
+	if err != nil {
+		return nil, status.Error(err, "Error determining the installation path")
+	}
+
+	_, err = exec.Command( //nolint:gosec // The user-controlled variables are sanitised above
+		"sh", "-c", "curl "+url+" | VERSION="+targetVersionString+" DESTDIR="+filepath.Dir(absolutePath)+" bash").CombinedOutput()
+	if err != nil {
+		return nil, status.Error(err, "Error upgrading subctl")
+	}
+
+	status.Success("Upgraded and installed subctl version: %s", to)
+
+	return &absolutePath, nil
+}
+
+func upgradeSubmariner(clusterInfo *cluster.Info, _ string, status reporter.Interface) error {
+	ctx := context.TODO()
+
+	status.Start("Starting upgrade process")
+	defer status.End()
+
+	brokerObj, brokerFound, err := getBroker(clusterInfo.RestConfig, constants.DefaultBrokerNamespace)
+	if err != nil {
+		return err
```

Comment:
So it ends the status.

```suggestion
		return status.Error(err, "")
```